diff --git a/node_modules/react-native/scripts/react_native_pods.rb b/node_modules/react-native/scripts/react_native_pods.rb
index 9cc53d8..79a3cf1 100644
--- a/node_modules/react-native/scripts/react_native_pods.rb
+++ b/node_modules/react-native/scripts/react_native_pods.rb
@@ -95,11 +95,18 @@ def use_react_native! (
   # Better to rely and enable this environment flag if the new architecture is turned on using flags.
   relative_path_from_current = Pod::Config.instance.installation_root.relative_path_from(Pathname.pwd)
   react_native_version = NewArchitectureHelper.extract_react_native_version(File.join(relative_path_from_current, path))
-  fabric_enabled = true
 
-  ENV['RCT_FABRIC_ENABLED'] = "1"
+  new_arch_flag = if ENV.key?('RCT_NEW_ARCH_ENABLED')
+    ENV['RCT_NEW_ARCH_ENABLED'] == '1'
+  else
+    new_arch_enabled
+  end
+
+  fabric_enabled = fabric_enabled || new_arch_flag
+
+  ENV['RCT_FABRIC_ENABLED'] = fabric_enabled ? "1" : "0"
   ENV['RCT_AGGREGATE_PRIVACY_FILES'] = privacy_file_aggregation_enabled ? "1" : "0"
-  ENV["RCT_NEW_ARCH_ENABLED"] = "1"
+  ENV["RCT_NEW_ARCH_ENABLED"] = new_arch_flag ? "1" : "0"
 
   prefix = path
 
